lib/portlayer/exec/container.go

```diff
@@ -119,6 +127,27 @@ func (c *Container) NewHandle() *Handle {
 	return newHandle(c)
 }
 
+// Refresh calls the propery collector to get config and runtime info
+func (c *Container) Refresh() error {
+	var o mo.VirtualMachine
+
+	// make sure we have vm
+	if c.vm == nil {
+		return fmt.Errorf("There is no backing VirtualMachine")
+	}
+	ctx := context.TODO()
+
+	if err := c.vm.Properties(ctx, c.vm.Reference(), []string{"config", "runtime"}, &o); err != nil {
+		return err
+	}
+
+	c.Config = o.Config
+	c.Runtime = &o.Runtime
+
```

Comment:
You are basically sync'ing the container's state with what the port layer has cached. Things like assigned ip addresses may have changed in the ExecutorConfig object on the container. These need to returned as output to several commands, like `network inspect` and `inspect`.